Code/Framework/AzToolsFramework/AzToolsFramework/Metadata/UuidUtils.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/Metadata/UuidUtils.h>
+#include <AzToolsFramework/Metadata/MetadataManager.h>
+#include <AzToolsFramework/Metadata/UuidEntry.h>
+
+namespace AzToolsFramework
+{
+    void UuidUtilComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<UuidUtilComponent, AZ::Component>();
+        }
+    }
+
+    bool UuidUtilComponent::CreateSourceUuid(AZ::IO::PathView absoluteFilePath, AZ::Uuid uuid)
+    {
+        auto* metadataInterface = AZ::Interface<IMetadataRequests>::Get();
+
+        if (!metadataInterface)
+        {
+            AZ_Assert(metadataInterface, "Programmer Error - IMetadataRequests interface is not available");
+            return false;
+        }
+
+        AzToolsFramework::UuidEntry entry;
+
+        if (metadataInterface->GetValue(absoluteFilePath, UuidKey, entry) && !entry.m_uuid.IsNull())
+        {
+            return false;
```

Comment:
Curious: Not quite sure how we want to handle this case. Do we want to override the old value and throw a warning/error here?

Reply:
I think it's best to not overwrite the old values because there shouldn't be a use case for programmatically changing the UUID since that'll break any references.